File: php/P0001.php
class Solution {

    /*
    * @param Integer[] $nums
    * @param Integer $target
    * @return Integer[]
    */
    function twoSum($nums, $target) {
        $map = array();
        
        foreach ($nums as $index => $num) {
            $complement = $target - $num;

            if (array_key_exists($complement, $map)) {
                return [$map[$complement], $index];
            }

            $map[$num] = $index;
        }

        // If no solution is found, return an empty array
        return array();
    }
}

File: dart/P0001.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> map = {};

    for (int i = 0; i < nums.length; i++) {
      int num = nums[i];
      int complement = target - num;

      if (map.containsKey(complement)) {
        return [map[complement]!, i];
      }

      map[num] = i;
    }

    // If no solution is found, return an empty list
    return [];
  }
}
